src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs

```diff
@@ -1477,24 +1477,25 @@ public async Task WaitForExitAsync(CancellationToken cancellationToken = default
                 }
 
                 // Wait until output streams have been drained
-                await WaitUntilOutputEOF().ConfigureAwait(false);
+                await WaitUntilOutputEOF().WaitAsync(cancellationToken).ConfigureAwait(false);
             }
             finally
             {
                 Exited -= handler;
             }
 
-            async ValueTask WaitUntilOutputEOF()
+            Task WaitUntilOutputEOF()
             {
-                if (_output != null)
+                if (_output is not null)
                 {
-                    await _output.WaitUntilEOFAsync(cancellationToken).ConfigureAwait(false);
+                    return _error is not null ?
+                        Task.WhenAll(_output.EOF, _error.EOF) :
```

Comment:
Nit: this is not cancelling the task created by `Task.WhenAll`.

Reply:
That's correct.  Just as previously it wasn't canceling the _readBufferTask.

Reply:
But perhaps it can be avoided by waiting sequentially?
```csharp
await _output.EOF.WaitAsync(token).ConfigureAwait(false);
await _error.EOF.WaitAsync(token).ConfigureAwait(false);
```

Reply:
What can be avoided?  Is your concern just that there's going to be an extra task object that remains as a dummy continuation on each EOF task until it completes?

Reply:
Yes.

Reply:
I'd explicitly thought about and decided it was ok.  But I don't feel strongly about it and will change it.